T7510: ospfd.frr.j2 ospf nssa translation error - fix template

ig0rb opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change summary
<!--- Provide a general summary of your changes in the Title above -->
Hi, I have updated ospf frr template to fix nssa translation

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
Jinja2 template patching as suggested in the forum.
[https://forum.vyos.io/t/ospf-area-type-nssa-translate-never/16593/8](url)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- optional: Link to related other tasks on Phabricator. -->
<!-- * https://vyos.dev/Txxxx -->
* https://vyos.dev/T7510
## Related PR(s)
<!-- Link here any PRs in other repositories that are required by this PR -->

## How to test / Smoketest result
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```

Or provide the output of the smoketest

```
$ /usr/libexec/vyos/tests/smoke/cli/test_xxx_feature.py
test_01_simple_options (__main__.TestFeature.test_01_simple_options) ... ok
```
-->
```
vyos@vyos:/tmp$ sh conf commands | grep ospf
set protocols ospf area 0.0.0.10 area-type nssa translate 'never'
vyos@vyos:/tmp$ vtysh -e "show run ospfd no-header"
frr version 10.2.2
frr defaults traditional
hostname localhost
log syslog
log facility local7
hostname vyos
service integrated-vtysh-config
!
router ospf
 auto-cost reference-bandwidth 100
 timers throttle spf 200 1000 10000
 area 0.0.0.10 nssa translate-never
exit
!
end
vyos@vyos:/tmp$ conf 
WARNING: You are currently configuring a live-ISO environment, changes will not persist until installed
[edit]
vyos@vyos# set protocols ospf area 0.0.0.10 area-type nssa translate always 
[edit]
vyos@vyos# commit
[edit]
vyos@vyos# exit
Warning: configuration changes have not been saved.
exit
vyos@vyos:/tmp$ vtysh -e "show run ospfd no-header"
frr version 10.2.2
frr defaults traditional
hostname localhost
log syslog
log facility local7
hostname vyos
service integrated-vtysh-config
!
router ospf
 auto-cost reference-bandwidth 100
 timers throttle spf 200 1000 10000
 area 0.0.0.10 nssa translate-always
exit
!
end
vyos@vyos:/tmp$ conf
WARNING: You are currently configuring a live-ISO environment, changes will not persist until installed
[edit]
vyos@vyos# del protocols ospf area 0.0.0.10 area-type nssa translate
[edit]
vyos@vyos# commit
[edit]
vyos@vyos# exit
Warning: configuration changes have not been saved.
exit
vyos@vyos:/tmp$ vtysh -e "show run ospfd no-header"
frr version 10.2.2
frr defaults traditional
hostname localhost
log syslog
log facility local7
hostname vyos
service integrated-vtysh-config
!
router ospf
 auto-cost reference-bandwidth 100
 timers throttle spf 200 1000 10000
 area 0.0.0.10 nssa
exit
!
end
vyos@vyos:/tmp$ 
```

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

ig0rb committed:
T7510: ospfd.frr.j2 ospf nssa translation error - fix template

c-po reviewed (changes requested):
Please also update the smoketests to check if the entry is available in Frr config when specified on the CLI. 

https://docs.vyos.io/en/latest/contributing/testing.html#manual-smoketest-run

ig0rb commented:
> Please also update the smoketests to check if the entry is available in Frr config when specified on the CLI.
> 
> https://docs.vyos.io/en/latest/contributing/testing.html#manual-smoketest-run

Hi, sorry, if my question look stupid, did I need to modify "test_protocols_ospf.py" to check the command ?

c-po commented:
> Hi, sorry, if my question look stupid, did I need to modify "test_protocols_ospf.py" to check the command ?

Yes, there you can add the code path to perform the smoketest. Set the CLI node and validate that it's present int the resulting FRR configuration.

c-po committed:
T7510: add smoketests for OSPF nssa translation

c-po committed:
T7510: add commit warnings about invalid use of OSPF area-types

To keep existing CLI behavior use a Warning() to prompt the user for an invalid
configuration. It is not possible to have more the one area-type defined per
area logically - the CLI does support it. In addition the backbone area cannot
be of type STUB or NSSA.

CLI configuration should be cleaned up using a migrator in the future.

c-po reviewed (approved):
Implemented smoketests:

```
$ /usr/libexec/vyos/tests/smoke/cli/test_protocols_ospf.py
test_ospf_18_area_translate_no_summary (__main__.TestProtocolsOSPF.test_ospf_0118_area_translate_no_summary) ... ok
test_ospf_01_defaults (__main__.TestProtocolsOSPF.test_ospf_01_defaults) ... ok
test_ospf_02_simple (__main__.TestProtocolsOSPF.test_ospf_02_simple) ... ok
test_ospf_03_access_list (__main__.TestProtocolsOSPF.test_ospf_03_access_list) ... ok
test_ospf_04_default_originate (__main__.TestProtocolsOSPF.test_ospf_04_default_originate) ... ok
test_ospf_05_options (__main__.TestProtocolsOSPF.test_ospf_05_options) ... ok
test_ospf_06_neighbor (__main__.TestProtocolsOSPF.test_ospf_06_neighbor) ... ok
test_ospf_07_redistribute (__main__.TestProtocolsOSPF.test_ospf_07_redistribute) ... ok
test_ospf_08_virtual_link (__main__.TestProtocolsOSPF.test_ospf_08_virtual_link) ... ok
test_ospf_09_interface_configuration (__main__.TestProtocolsOSPF.test_ospf_09_interface_configuration) ... ok
test_ospf_11_interface_area (__main__.TestProtocolsOSPF.test_ospf_11_interface_area) ... ok
test_ospf_12_vrfs (__main__.TestProtocolsOSPF.test_ospf_12_vrfs) ... ok
test_ospf_13_export_list (__main__.TestProtocolsOSPF.test_ospf_13_export_list) ... ok
test_ospf_14_segment_routing_configuration (__main__.TestProtocolsOSPF.test_ospf_14_segment_routing_configuration) ... ok
test_ospf_15_ldp_sync (__main__.TestProtocolsOSPF.test_ospf_15_ldp_sync) ... ok
test_ospf_16_graceful_restart (__main__.TestProtocolsOSPF.test_ospf_16_graceful_restart) ... ok
test_ospf_17_duplicate_area_network (__main__.TestProtocolsOSPF.test_ospf_17_duplicate_area_network) ... ok

----------------------------------------------------------------------
Ran 17 tests in 271.259s

OK
```

dmbaturin reviewed (approved):
A straightforward addition, covered by smoke tests. No objections.
